lib/Cake/Controller/Component/FlashComponent.php

```diff
@@ -82,12 +82,18 @@ public function set($message, $options = array()) {
 		}
 		$options['element'] = $plugin . 'Flash/' . $element;
 
-		CakeSession::write('Message.' . $options['key'], array(
+		$messages = CakeSession::read('Message.' . $options['key']);
+
+		$newMessage = array(
```

Comment:
In 3.x we support a `clear` option to allow for the flash message stack to be reset/cleared when adding a new message. Could we also have that option in 2.x? Without it maintaining existing behavior is not possible for those who want/need it.